src/common/domain/README.md

```diff
@@ -0,0 +1,5 @@
+# Dynamically generated journal interface goes here
+
+The type information for the journal should live here as `Journal.d.ts` - if you don't have this file it can by generated by doing `npm run bootstrap`.
+
```

Comment:
This is really cool! I would consider putting generated code in a __generated__ sub directory so people don't end up editing it, and if we hand crank models in the future generated code isn't all mixed up with them.

Reply:
Super cool!

Reply:
Will chase getting ourselves an NPM package repo to put this schema into, then it can have a pipeline/versioning scheme just like everything else.

Reply:
We could use git submodules if the NPM repo isn't happening

Reply:
I think that we probably want to extract the schemas (and possibly the generated mode files) to a separate npm module that we could share across multiple projects (mobile app, other microservices etc). 

This would be essentially our rest contract between frontend and backend - the single source of truth  + reduces the duplication of code between projects (we already see those files copied over between mobile app and the microservice.

Plus it would be nice to be able to version the schemas at some point and be able to compare/test a different version of the schema (eg. mobile app still using v1 where API expects v2).

Assuming that the schema will be iterated over to improve our strict typing of the api responses that would help us with not needing to regenerate the interfaces each time we're building the microservice...